File: bindings/csharp/tests/STRling.Tests/UsPhoneTest.cs
using Xunit;
using Strling.Simply;

namespace STRling.Tests;

public class UsPhoneTest
{
    [Fact]
    public void FluentApiProducesExpectedRegexString()
    {
        var phone = S.Merge(
            S.Start(),
            S.Digit(3).Capture(),
            S.AnyOf("-. ").May(),
            S.Digit(3).Capture(),
            S.AnyOf("-. ").May(),
            S.Digit(4).Capture(),
            S.End()
        );

        var regex = phone.Compile();
        System.IO.File.WriteAllText("/tmp/us_phone_regex.txt", regex);

        Assert.Equal(@"^(\d{3})[-. ]?(\d{3})[-. ]?(\d{4})$", regex);
    }
}

File: bindings/cpp/test/unit/us_phone_test.cpp
/* us_phone_test.cpp — verifies simply API builds the expected US phone regex */
#include <gtest/gtest.h>
#include "strling/simply.hpp"

using namespace strling::simply;

TEST(SimplyPattern, USPhoneBuilderMatchesReference) {
    // Build the phone example exactly like the README/target API
    auto phone = merge({
        start(),
        digit(3).as_capture(),
        any_of("-. ").may(),
        digit(3).as_capture(),
        any_of("-. ").may(),
        digit(4).as_capture(),
        end()
    });

    std::string actual = phone.compile();

    // Expected builder-style optimized pattern (positional captures + optional separators)
    std::string expected = "^(\\d{3})[-. ]?(\\d{3})[-. ]?(\\d{4})$";

    EXPECT_EQ(actual, expected);
}
